src/main/java/io/vertx/ext/jdbc/spi/JDBCDecoder.java

```diff
@@ -0,0 +1,76 @@
+package io.vertx.ext.jdbc.spi;
+
+import io.vertx.ext.jdbc.impl.actions.JDBCTypeProvider;
+import io.vertx.ext.jdbc.impl.actions.SQLValueProvider;
+
+import java.sql.CallableStatement;
+import java.sql.JDBCType;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Represents for JDBC decoder from SQL value to Java value
+ * <p>
+ * The default decoder provides the best efforts to convert {@code SQL type} to standard {@code Java type} as {@code JDBC 4.2} spec.
+ * <p>
+ * You can replace it to adapt to a specific SQL driver by creating your owns then includes in the SPI file
+ * ({@code META-INF/services/io.vertx.ext.jdbc.spi.JDBCDecoder})
+ *
+ * @see <a href=https://docs.oracle.com/javase/8/docs/technotes/guides/jdbc/jdbc_42.html">Mapping of java.sql.Types to SQL types</a>
```

Comment:
The final `"` at the end of the url is breaking the syntax highlighter and will also create a broken link reference in the javadoc.